cmd/launch.go

```diff
@@ -215,16 +215,7 @@ func launchInteractive(h *ec2helper.EC2Helper) {
 	}
 
 	// Launch On-Demand or Spot instance based on capacity type
-	if simpleConfig.CapacityType == question.CapacityTypes.OnDemand {
-		_, err = h.LaunchInstance(simpleConfig, detailedConfig, confirmation == cli.ResponseYes)
-	} else {
-		err = LaunchSpotInstance(h, simpleConfig, detailedConfig, confirmation)
-	}
-
-	if cli.ShowError(err, "Launching instance failed") {
-		return
-	}
-	ReadSaveConfig(simpleConfig)
```

Comment:
`ReadSaveConfig()` should stay at this level, rather than going into the `LaunchCapacityInstance()` function, because it's not directly related to launching the instance per se.

In addition, `LaunchCapacityInstance()` should return its error message and let the caller handle it how they want (in this case, the `cli.ShowError()` stuff).